src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilder.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Identity.Client;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Authentication builder specific for Microsoft identity platform.
+    /// </summary>
+    public class MicrosoftIdentityWebAppAuthenticationBuilder : MicrosoftIdentityBaseAuthenticationBuilder
+    {
+        /// <summary>
+        ///  Constructor.
+        /// </summary>
+        /// <param name="services"> The services being configured.</param>
+        /// <param name="openIdConnectScheme">Default scheme used for OpenIdConnect.</param>
+        /// <param name="configureMicrosoftIdentityOptions">Action called to configure
+        /// the <see cref="MicrosoftIdentityOptions"/>Microsoft identity options.</param>
+        /// <param name="configurationSection">Optional configuration section.</param>
+        internal MicrosoftIdentityWebAppAuthenticationBuilder(
+            IServiceCollection services,
+            string openIdConnectScheme,
+            Action<MicrosoftIdentityOptions> configureMicrosoftIdentityOptions,
+            IConfigurationSection? configurationSection)
+            : base(services, configurationSection)
+        {
+            OpenIdConnectScheme = openIdConnectScheme;
+            ConfigureMicrosoftIdentityOptions = configureMicrosoftIdentityOptions;
+
+            if (ConfigureMicrosoftIdentityOptions == null)
+            {
+                throw new ArgumentNullException(nameof(configureMicrosoftIdentityOptions));
+            }
```

Comment:
Move the null check infront?
```suggestion
            if (configureMicrosoftIdentityOptions == null)
            {
                throw new ArgumentNullException(nameof(configureMicrosoftIdentityOptions));
            }
            OpenIdConnectScheme = openIdConnectScheme;
            ConfigureMicrosoftIdentityOptions = configureMicrosoftIdentityOptions;
```

Reply:
i didn't look at this closely enough and made the change, but it doesn't work as we have to set the options and then check if they are null after, so have left it as is.. let's use the "suggested changes" features for xml comments only, not code changes, unless you've tested and know it doesn't break things. sound good?